### ISSUE TITLE
Watch files and sync local<->remote

### ISSUE THREAD

**Issue by OscarGodson** [lukasz-wronski/vscode-ftp-sync#49]
I'd love to have it so that when files change they will auto be pushed up even if i don't save the file. Most common use case is after a `git pull` I'd like to not have to manually sync. Instead it sees the files changed and syncs on the server.

**Comment by HansLanger** [comment]
Agree, this is more like a regular FTP than a SYNC files. It should sync on save changed file.

**Comment by abderrahmane-tj** [comment]
I agree with @OscarGodson, syncing on actual file change should be an option too. Other scenarios are : running a transpiler in the background that would change files, and is something i want to be pushed

**Comment by OscarGodson** [comment]
@abderrahmane-tj yep! Right now I push the files up and have a file watcher on the server that builds, but would _much_ rather prefer files built locally.
